src/python/pants/backend/scala/subsystems/scala.py

```diff
@@ -44,6 +44,17 @@ class ScalaSubsystem(Subsystem):
         advanced=True,
     )
 
+    add_siblings_as_dependencies = BoolOption(
+        "--add-siblings-as-dependencies",
+        default=True,
```

Comment:
@stuhood: Should the default here still be `True` as in the existing code? Enabling this option would actually be a fallback if dependency inference failed to infer the right dependencies.

Reply:
See my proposal at https://github.com/pantsbuild/pants/issues/14382#issuecomment-1154528470. Imo this probably should not exist as an option long term, and we only need it to avoid breaking our deprecation policy

Reply:
@Eric-Arellano @tdyas let me know what is the name, the default option, and where to place the option (should I move it to the scala infer subsystem?)

Reply:
My proposal:

1. Move to `scala-infer`
2. Call it `--force-add-siblings-as-dependencies`
3. Update `help` to clarify that if this option is not set, then sibling dependencies will only be added if `[scala-infer].imports` is disabled.
4. Deprecate this option not being explicitly set so that we can change the default in a future Pants release.

That's pretty involved...I suggest this change is kept for a dedicated PR.

Reply:
@Eric-Arellano @tdyas here is the new PR https://github.com/pantsbuild/pants/pull/15841

Removed the commit from this one.